src/frontend/src/optimizer/rule/index_delta_join_rule.rs

```diff
@@ -105,8 +111,10 @@ impl Rule for IndexDeltaJoinRule {
             None
         }
 
-        if let Some(left) = match_indexes(&left_indices, input_left) {
-            if let Some(right) = match_indexes(&right_indices, input_right) {
+        // Delta join only needs to backfill one stream flow.
+        // Here we choose the left one to backfill and right one to pure chain.
```

Comment:
I'd like to point out that `Chain` means the concatenation of the "snapshot" and the "upstream", not the connection between two tables. But it seems more complicated to introduce another concept, it's okay for me to make it one of the "generalized" chains, but should be like an "upstream-only" chain instead of a "pure" chain. 😂

Reply:
According to the discussion, I create a new chain type `UpstreamOnly` which only consumes the upstream messages.